zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/SchemaManager.java

```diff
@@ -124,20 +130,31 @@ private void initialiseIndices() {
     }
 
     final var existingIndexNames =
-        searchEngineClient.getMappings(allIndexNames(), MappingSource.INDEX).keySet();
+        Collections.synchronizedSet(
+            searchEngineClient.getMappings(allIndexNames(), MappingSource.INDEX).keySet());
 
     LOG.info(
         "Found '{}' existing indices. Create missing index templates based on '{}' descriptors.",
         existingIndexNames.size(),
         indexTemplateDescriptors.size());
-    indexDescriptors.stream()
-        .filter(descriptor -> !existingIndexNames.contains(descriptor.getFullQualifiedName()))
-        .forEach(
-            descriptor -> {
-              LOG.info("Create missing index '{}'", descriptor.getFullQualifiedName());
-              searchEngineClient.createIndex(
-                  descriptor, getIndexSettingsFromConfig(descriptor.getIndexName()));
-            });
+    final var futures =
+        indexDescriptors.stream()
+            .filter(descriptor -> !existingIndexNames.contains(descriptor.getFullQualifiedName()))
+            .map(
+                descriptor ->
+                    // run creation of indices async as virtual thread
+                    CompletableFuture.runAsync(
+                        () -> {
+                          LOG.info("Create missing index '{}'", descriptor.getFullQualifiedName());
+                          searchEngineClient.createIndex(
+                              descriptor, getIndexSettingsFromConfig(descriptor.getIndexName()));
+                        },
+                        virtualThreadExecutor))
+            .toArray(CompletableFuture[]::new);
+
+    // We need to wait for the completion, to make sure all indices has been created successfully
+    // Doing this in parallel is still speeding up the bootstrap time
+    CompletableFuture.allOf(futures).join();
```

Comment:
:wrench: I think it's better to always put a timeout in these occasion, probably in the order of 1 minute or more. Just to not have something waiting indefinitely, but just throw and exception and maybe restart the pod?

Reply:
Oh yes, I somehow oversaw this. Thanks for spotting!